### PR TITLE
Implement a better error manager and debug

### LINKED ISSUE #173: Implement a better error manager and debug
**Is your feature request related to a problem? Please describe.**
When the error are showing in the library display not relevant information, improve the information displaying

### PR DESCRIPTION
**Pull request by Ivans51**
Signed-off-by: Ivan Del Pino <idelpino@teclib.com>

### Changes description

Handle error in:
 - Battery
 - Bluetooth
 - Camera
 - Controllers
 - CPU
 - Drives
 - Hardware
 - Inputs
 - Locations
 - Memory
 - Networks
 - Operating system
 - Phone Status
 - Sensor
 - SIM Card
 - Software
 - Storege
 - Usb
 - User
 - Category
 - Categories

### Checklist

Please check if your PR fulfills the following specifications:

- [x] Tests for the changes have been added
- [x] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@ivans51 |12|
|@rafaelje |2|

### References

Closes: https://github.com/flyve-mdm/android-inventory-library/issues/173
Related #N/A
Depends on #N/A

### CODE REVIEW

**Review by ajsb85 — APPROVED** [review]

File: inventory/src/main/java/org/flyve/inventory/CommonErrorType.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    public static final int CATEGORY_TO_XML = 2301;
+    public static final int CATEGORY_TO_XML_WITHOUT_PRIVATE = 2302;
+    public static final int CATEGORY_TO_JSON = 2303;
+    public static final int CATEGORY_TO_JSON_WITHOUT_PRIVATE = 2304;
+    public static final int CATEGORIES_TO_XML = 2305;
+    public static final int CATEGORIES_TO_XML_WITHOUT_PRIVATE = 2306;
+    public static final int CATEGORIES_TO_JSON = 2307;
+    public static final int CATEGORIES_TO_JSON_WITHOUT_PRIVATE = 2308;
+
+    /* Others */
```
Comment (rafaelje) [inline inventory/src/main/java/org/flyve/inventory/CommonErrorType.java]: Avoid use other classification

**Review by ajsb85 — APPROVED** [review]

File: inventory/src/main/java/org/flyve/inventory/CommonErrorType.java [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    public static final int BIOS_BOARD_MANUFACTURER = 204;
+    public static final int BIOS_MOTHER_BOARD_MODEL = 205;
+    public static final int BIOS_TAG = 206;
+    public static final int BIOS_MOTHER_BOARD_SERIAL = 207;
+    public static final int BIOS_SYSTEM_SERIAL = 208;
+    public static final int BIOS_SERIAL_PRIVATE = 209;
+    public static final int BIOS_CPU_SERIAL = 210;
+
+    /* Bluetooth */
+
```
Comment (rafaelje) [inline inventory/src/main/java/org/flyve/inventory/CommonErrorType.java]: Remove this space

File: inventory/src/main/java/org/flyve/inventory/Utils.java [hunk 3]
```diff
@@ -310,8 +310,8 @@ public static String getCatInfoMultiple(String path) {
             while (s.hasNextLine()) {
                 concatInfo.append(s.nextLine());
             }
-        } catch (Exception e) {
-            FILog.e(e.getMessage());
+        } catch (Exception ex) {
+            FILog.e(CommonErrorType.UTILS_CAT_INFO_MULTIPLE + " " + ex.getMessage());
```
Comment (rafaelje) [inline inventory/src/main/java/org/flyve/inventory/Utils.java]: Use XML string to create this message instead string concatenate

File: inventory/src/main/java/org/flyve/inventory/Utils.java [hunk 4]
```diff
@@ -320,8 +320,8 @@ public static String getCatInfo(String path) {
         try {
             Scanner s = new Scanner(new File(path));
             return s.next();
-        } catch (Exception e) {
-            FILog.e(e.getMessage());
+        } catch (Exception ex) {
+            FlyveLog.e(CommonErrorType.UTILS_CAT_INFO + " " + ex.getMessage());
```
Comment (rafaelje) [inline inventory/src/main/java/org/flyve/inventory/Utils.java]: Implement XML resources instead concatenation string

**Review by rafaelje — CHANGES_REQUESTED** [review]

**Review by rafaelje — APPROVED** [review]

**Pull request merged**